geo/src/algorithm/coordinate_position.rs

```diff
@@ -0,0 +1,749 @@
+            line_string.calculate_coordinate_position(coord, is_inside, boundary_count);
+        }
+    }
+}
+
+impl<T> CoordinatePosition<T> for MultiPolygon<T>
+where
+    T: HasKernel,
+{
+    fn calculate_coordinate_position(
+        &self,
+        coord: &Coordinate<T>,
+        is_inside: &mut bool,
+        boundary_count: &mut usize,
+    ) {
+        for polygon in &self.0 {
+            polygon.calculate_coordinate_position(coord, is_inside, boundary_count);
+        }
+    }
+}
+
+impl<T> CoordinatePosition<T> for GeometryCollection<T>
+where
+    T: HasKernel,
+{
+    fn calculate_coordinate_position(
+        &self,
+        coord: &Coordinate<T>,
+        is_inside: &mut bool,
+        boundary_count: &mut usize,
+    ) {
+        for geometry in self {
+            geometry.calculate_coordinate_position(coord, is_inside, boundary_count);
+        }
+    }
+}
+
+impl<T> CoordinatePosition<T> for Geometry<T>
+where
+    T: HasKernel,
+{
+    fn calculate_coordinate_position(
+        &self,
+        coord: &Coordinate<T>,
+        is_inside: &mut bool,
+        boundary_count: &mut usize,
+    ) {
+        if self.is_empty() {
+            return;
+        }
+
+        match self {
+            Geometry::Point(point) => {
+                point.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::Line(line) => {
+                line.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::LineString(line_string) => {
+                line_string.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::Polygon(polygon) => {
+                polygon.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::Rect(rect) => {
+                rect.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::Triangle(triangle) => {
+                triangle.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::MultiPoint(multi_point) => {
+                multi_point.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::MultiLineString(multi_line_string) => {
+                multi_line_string.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::MultiPolygon(multi_polygon) => {
+                multi_polygon.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+            Geometry::GeometryCollection(geometry_collection) => {
+                geometry_collection.calculate_coordinate_position(coord, is_inside, boundary_count)
+            }
+        }
+    }
+}
+
+/// Calculate the position of a `Coordinate` relative to a
+/// closed `LineString`.
+pub fn coord_pos_relative_to_ring<T>(coord: Coordinate<T>, linestring: &LineString<T>) -> CoordPos
```

Comment:
This method was moved from utils unmodified.

In an effort to keep `utils` from becoming a big junk drawer, this seemed like a more specific home for it.